components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilderTest.java

```diff
@@ -154,17 +228,293 @@ public void setUp() throws Exception {
+        mockRealmService();
+        messageContext = getTokenReqMessageContextForUser(getDefaultAuthenticatedLocalUser(), invalidClientId);
+        try {
+            String authoIDToken = defaultIDTokenBuilder.buildIDToken(messageContext, tokenRespDTO);
+            JWTClaimsSet claims = SignedJWT.parse(authoIDToken).getJWTClaimsSet();
+            Assert.assertEquals(claims.getAudience().get(0), CLIENT_ID);
+            Assert.assertEquals(claims.getIssuer(), "https://localhost:9443/oauth2/token");
+        } catch (IdentityOAuth2Exception e) {
+            Assert.assertEquals(e.getMessage(),
+                    "Error occurred while getting app information for client_id: " +
+                            invalidClientId);
+        }
+    }
+
+    @Test
+    public void testClientIDNotFoundExceptionForAuthorization() throws Exception {
+
+        String invalidClientId = "3f8b04332794d8dabfba9390aa423614";
+        mockRealmService();
+        OAuthAuthzReqMessageContext oAuthAuthzReqMessageContext = getOAuthAuthzReqMessageContextForUser
+                (getDefaultAuthenticatedLocalUser(), invalidClientId);
+        OAuth2AuthorizeRespDTO oAuth2AuthorizeRespDTO = new OAuth2AuthorizeRespDTO();
+        oAuth2AuthorizeRespDTO.setAccessToken("2sa9a678f890877856y66e75f605d456");
+        try {
+            String idToken = defaultIDTokenBuilder
+                    .buildIDToken(oAuthAuthzReqMessageContext, oAuth2AuthorizeRespDTO);
+            JWTClaimsSet claims = SignedJWT.parse(idToken).getJWTClaimsSet();
+            Assert.assertEquals(claims.getAudience().get(0), invalidClientId);
+            Assert.assertEquals(claims.getIssuer(), "https://localhost:9443/oauth2/token");
+        } catch (IdentityOAuth2Exception e) {
+            Assert.assertEquals(e.getMessage(),
+                    "Error occurred while getting app information for client_id: " +
+                            invalidClientId);
+        }
+    }
+
+    private OAuthAppDO getOAuthAppDO(String algorithm) throws Exception {
+
+        OAuthAppDO entry = new OAuthAppDO();
+        entry.setOauthConsumerKey(CLIENT_ID);
+        entry.setOauthConsumerSecret("87n9a540f544777860e75f605d435");
+        entry.setApplicationName("myApp");
+        entry.setCallbackUrl(TestConstants.CALLBACK);
+        entry.setOauthVersion("OAuth-2.0");
+        entry.setState("ACTIVE");
+        entry.setUserAccessTokenExpiryTime(3600000);
+        entry.setApplicationAccessTokenExpiryTime(3600000);
+        entry.setRefreshTokenExpiryTime(84600000);
+        entry.setAppOwner(user);
+        entry.setIdTokenEncryptionEnabled(true);
+        entry.setIdTokenEncryptionAlgorithm(algorithm);
+        entry.setIdTokenEncryptionMethod("A128GCM");
+        return entry;
+    }
+
+    private EncryptedJWT decryptToken (String  token) throws Exception {
+
+        InputStream file = new FileInputStream("src/test/resources/keyStore/testkeystore.jks");
```

Comment:
can we have this keystore in a separate folder?  (to understand which keystore this is and why it is used)


```suggestion
        InputStream file = new FileInputStream("src/test/resources/keyStore/encryption/appKeystore.jks");
```

Reply:
Addressed with 0c791e543d284b9875bbb6847edb5bbe9a1d82e1